neuralmonkey/decoders/beam_search_decoder.py

```diff
@@ -248,65 +270,84 @@ def body(*args) -> BeamSearchLoopState:
             logprobs = unfinished_logprobs + finished_logprobs
 
             # update hypothesis scores
-            # shape(hyp_probs) = beam x vocabulary
+            # shape(hyp_probs) = (batch * beam) x vocabulary
             hyp_probs = tf.expand_dims(bs_state.logprob_sum, 1) + logprobs
 
             # update hypothesis lengths
             hyp_lengths = bs_state.lengths + 1 - tf.to_int32(bs_state.finished)
 
-            # shape(scores) = beam x vocabulary
+            # shape(scores) = (batch * beam) x vocabulary
             scores = hyp_probs / tf.expand_dims(
                 self._length_penalty(hyp_lengths), 1)
 
-            # flatten so we can use top_k
-            scores_flat = tf.reshape(scores, [-1])
+            # reshape to batch_size x (beam_size * vocabulary) for topk
+            input_beam_size = bs_state.input_beam_size
+            scores_flat = tf.reshape(
+                scores, [-1, input_beam_size * len(self.vocabulary)])
 
             # shape(both) = beam
             topk_scores, topk_indices = tf.nn.top_k(
                 scores_flat, self._beam_size)
-
-            topk_scores.set_shape([self._beam_size])
```

Comment:
kam se poděl tenhle command?

Reply:
Zahodil jsem ho, je skutecne nezbytny (ten shape je garantovany tou funkci top_k, ne)??

Reply:
Zkus si pustit pudb a koukni se na get_shape. To samý je ale potřeba udělat i s tim `topk_indices`, takhle bych to určitě nenechával. Buď obojí vyhodit nebo obojí nechat.

Reply:
Podle mě get_shape nebude vědět, že je to beam size, ale bude tam otazníček nebo none.

Reply:
Tak jsem ti ho tam pridal.